mobile/lib/features/wallet/send/confirm_payment_modal.dart

```diff
@@ -27,45 +33,110 @@ void showConfirmPaymentModal(BuildContext context, Destination destination, Amou
       builder: (BuildContext context) {
         return SingleChildScrollView(
             child: SizedBox(
-                height: 320,
+                height: 420,
                 child: Scaffold(
                     body: ConfirmPayment(
+                  payWithUsdp: payWithUsdp,
                   destination: destination,
-                  amount: amount,
+                  sats: sats,
+                  usdp: usdp,
                 ))));
       });
 }
 
 class ConfirmPayment extends StatelessWidget {
   final Destination destination;
-  final Amount? amount;
+  final bool payWithUsdp;
+  final Amount sats;
+  final Amount usdp;
 
-  const ConfirmPayment({super.key, required this.destination, this.amount});
+  const ConfirmPayment(
+      {super.key,
+      required this.destination,
+      required this.payWithUsdp,
+      required this.sats,
+      required this.usdp});
 
   @override
   Widget build(BuildContext context) {
     final walletService = context.read<WalletChangeNotifier>().service;
+    final submitOderChangeNotifier = context.read<SubmitOrderChangeNotifier>();
+    final formatter = NumberFormat("#,###,##0.00", "en");
 
-    final amt = destination.amount.sats > 0 ? destination.amount : amount!;
+    final tradeValuesChangeNotifier = context.watch<TradeValuesChangeNotifier>();
+
+    final tradeValues = tradeValuesChangeNotifier.fromDirection(Direction.long);
+    tradeValues.updateLeverage(Leverage(1));
```

Comment:
Why do you need to update the leverage?

Reply:
because the default leverage is 2 and if I wouldn't set it to 1 we would get an incorrect sats amount from the quantity.